chatbot/chatbot_agent.py

```diff
@@ -0,0 +1,318 @@
+## Import Python Packages
+import os
+import requests
+from typing import List
+
+import openai
+import chromadb
+import langchain
+
+from langchain.embeddings.openai import OpenAIEmbeddings
+from langchain.vectorstores import Chroma
+from langchain.text_splitter import TokenTextSplitter
+from langchain.llms import OpenAI
+from langchain.chains import ChatVectorDBChain
+from langchain.chains import RetrievalQA
+
+from langchain.document_loaders import GutenbergLoader
+from langchain.docstore.document import Document
+from langchain.document_loaders.base import BaseLoader
+
+from langchain.prompts import PromptTemplate
+from langchain.chains.qa_with_sources import load_qa_with_sources_chain
+from langchain.chains.question_answering import load_qa_chain
+
+
+
+
+## chatbot agent class
+class ChatbotAgent:
+    def __init__(self, _openai_api_key: str, _sources_urls: List[str], _persist_directory: str):

```

Comment:
Remove the _ prefix from the function parameter name. We usually use _ to differentiate a private class member in Python. Apply this to other places if possible.

Reply:
Done